include/ck/tensor_operation/gpu/grid/gridwise_batched_gemm_bias_gelu_gemm_bias_xdl_cshuffle_v1.hpp

```diff
@@ -0,0 +1,1231 @@
+        constexpr auto n2 = acc0_thread_desc_m0_n0_m1_n1_m2_n2_n3_n4.GetLength(I5);
+        constexpr auto n3 = acc0_thread_desc_m0_n0_m1_n1_m2_n2_n3_n4.GetLength(I6);
+        constexpr auto n4 = acc0_thread_desc_m0_n0_m1_n1_m2_n2_n3_n4.GetLength(I7);
+
+        constexpr auto b1_block_slice_copy_step = make_multi_index(Gemm1KPerBlock / B1K1, 0, 0);
+
+        // d0 matrix threadwise copy
+        constexpr auto d0_thread_desc_m0_n0_m1_n1_m2_n2_m3_n3_n4_n5 =
+            make_naive_tensor_descriptor_packed(make_tuple(I1,   // MBlockId
+                                                           I1,   // NBlockID
+                                                           I1,   // MRepeat
+                                                           I1,   // NRepeat
+                                                           I1,   // MWaveId
+                                                           I1,   // NWaveId
+                                                           I1,   // MPerXdl
+                                                           I1,   // NGroupNum
+                                                           I1,   // NInputNum
+                                                           n4)); // registerNum
+
+        StaticBuffer<AddressSpaceEnum::Vgpr,
+                     A0B0B1DataType,
+                     d0_thread_desc_m0_n0_m1_n1_m2_n2_m3_n3_n4_n5.GetElementSpaceSize(),
+                     true>
+            d0_thread_buf;
+        const auto wave_id     = GetGemm0WaveIdx();
+        const auto wave_m_n_id = GetGemm0WaveMNIdx(wave_id[I2]); // I2: 0~63
+
+        constexpr auto acc0_thread_desc = make_naive_tensor_descriptor_packed(
+            make_tuple(Number<Gemm0MXdlPerWave>{}, Number<Gemm0NXdlPerWave>{}, n2, n4));
+
+#if 0
+        const index_t block_id  = get_block_1d_id();
+        const index_t thread_id = get_thread_local_1d_id();
+        if(block_id == 11)
+        {
+            printf("block id: %d  block_work_idx[I0]: %d block_work_idx[I1]: %d ,thread id: %d, "
+                   "wave id :{%d %d %d} "
+                   "wave_m_n_id: {%d %d}\n",
+                   block_id,
+                   block_work_idx[I0],
+                   block_work_idx[I1],
+                   thread_id,
+                   wave_id[I0],
+                   wave_id[I1],
+                   wave_id[I2],
+                   wave_m_n_id[I0],
+                   wave_m_n_id[I1]);
+        }
+
+#endif
+        auto d0_threadwise_copy =
+            ThreadwiseTensorSliceTransfer_v2<A0B0B1DataType,
+                                             A0B0B1DataType,
+                                             decltype(d0_griddesc_m0_n0_m1_n1_m2_n2_m3_n3_n4_n5),
+                                             decltype(d0_thread_desc_m0_n0_m1_n1_m2_n2_m3_n3_n4_n5),
+                                             Sequence<I1, I1, I1, I1, I1, I1, I1, I1, I1, n4>,
```

Comment:
Have you tried copying bigger slice than `Sequence<I1, I1, I1, I1, I1, I1, I1, I1, I1, n4>` for each threadwise_copy.run()? That could help you avoid the quadruple nested static_for loop down below.

Reply:
I'll try it

Reply:
@rosenrodt From the test results, there is no obvious improvement

Reply:
Yes it's just for readability.

Reply:
As discussed with @ltqin, trying bigger slice is of no perf gain and of lower priority adjustment